ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -23,23 +24,30 @@ def metric_names(payload, prefix, sufix, **kwargs):
     special_label = kwargs.get('special_label')
     for entry in payload:
         if special_label == 'fan':
-            e = re.sub(r'[\d].*$', '', entry.lower())
-            e = re.sub(r'[\(\)]', '', e).split()
+            # NOTE (iurygregory): regex to remove a sequence of numbers and
+            # letters that comes after the fan sensor name.
+            # e.g: 'Fan4B (0x43)' will be 'fan (0x43)'
+            e = re.sub(r'fan\d*[a-z]*', 'fan', entry.lower())
+            # NOTE (iurygregory): regex to remove brackets and it content
+            # e.g: 'fan (0x43)' will turn into ['fan']
+            e = re.sub(r'\(.*\)', '', e).split()
```

Comment:
from the example you provided, wouldn't assigning ['fan'] to e be sufficient ?

Reply:
I can't put only "fan" because there is an entry called: "Fan Redundancy (0x78)" https://github.com/metal3-io/ironic-prometheus-exporter/blob/master/ironic_prometheus_exporter/tests/data2.json#L1072